postcommit.py

```diff
@@ -0,0 +1,642 @@
+				result = parent_rev_pattern.findall(str(parent_rev))
+				if(result[0] != "parent"):
+					cmd = "git log --numstat " + changeset[0] + pretty_format
+				else:
+					cmd = "git log --numstat " +  changeset[length-1] + '~1..' + changeset[0] + pretty_format
+		elif newId == "0000000000000000000000000000000000000000":
+			cmd = ""			# For a branch deletion command statement is empty
+		else:
+			cmd = "git log --numstat " + oldId + '..' +  newId + pretty_format
+        	commits = status_check(cmd, self.module_dir)
+        	self.parseCommits(str(commits[0]), branchname)
+
+	
+#	***************************************************************************************************************
+#	updateMergeDetails :
+#
+#			This method used to find how many files added,removed,modified in single changeset.Also it 
+#			shows number of added and deleted lines in decimal notation for each files. 
+#
+#			@param : log
+#			@param : parents
+#			@param : parent_short
+#			
+#			@return : log         	
+#					
+#	*************************************************************************************************************** 
+    	def updateMergeDetails(self, log, parents, parents_short):
+        	log['is_merge'] = 1
+        	log['merged_by'] = log['user']
+        	log['parent2_branch'] = parents[1]
+        	log['parent2']  = parents[1]
+        	log['parent2_short']  = parents_short[1]
+        	return log
+
+	
+#	***************************************************************************************************************
+#	filestatus :
+#
+#			This method used to find how many files added,removed,modified in single changeset.Also it 
+#			shows number of added and deleted lines in decimal notation for each files. 
+#
+#			@param : changeset
+#			
+#			@return : {'Removed': [], 'Added': [], 'Modified': [{'AddedLines': '1', 'File': 'test', 
+#				'RemovedLines':'2'}]}         						
+#	*************************************************************************************************************** 
+    	def filestatus(self, changeset):
+        	cmd = "git show --name-status --pretty='format:CMHash:%H' " + changeset
+        	commit_info = status_check(cmd, self.module_dir)
+        	commit_sts_pattern = re.compile('CMHash:([a-z0-9]+)\n+(([AMD]\s+(.*?)\n)*)?')
+        	commit_sts_detail = commit_sts_pattern.findall(str(commit_info[0]))
+
+        	result = {'Modified':[], 'Added': [], 'Removed':[]}
+
+		cmd = "git show --numstat --pretty='format:CMHash:%H' " + changeset
+		filter_info = status_check(cmd,self.module_dir)
+		filter_sts_pattern = re.compile('CMHash:([a-z0-9]+)\n+(([0-9\-]+\s+[0-9\-]+\s+(.*?)\n)*)?')
+		filter_sts_detail = filter_sts_pattern.findall(str(filter_info[0]))
+
+		if commit_sts_detail:
+           		commit_sts_detail = commit_sts_detail[0]
+			filter_sts_detail = filter_sts_detail[0] 
+
+		if commit_sts_detail:
+            		if (commit_sts_detail[0] == changeset and commit_sts_detail[1] != ''):
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS1066)